### PR TITLE
vkreplay: Add an option to quit on any error

### PR DESCRIPTION
**Pull request by justuswang**
This change adds an option "-x" to vkreplay to make vkreplay quit on
any error when replaying a trace file.

It helps user to find the first replay error much easier when replaying with "quit on any error" enabled. 
 Because vkreplay will quit on the first error instead of printing lots of error messages before quit.

### CODE REVIEW

File: vktrace/vktrace.md [hunk 1]
```diff
@@ -111,6 +111,7 @@ The  `vkreplay` command-line  options are:
 | -c&nbsp;&lt;bool&gt;<br>&#x2011;&#x2011;CompatibilityMode&nbsp;&lt;bool&gt; | Enable compatibility mode - modify api calls as needed when replaying trace file created on different platform than replay platform. For example: Convert trace file memory indices to replay device memory indices. | true |
 | -s&nbsp;&lt;string&gt;<br>&#x2011;&#x2011;Screenshot&nbsp;&lt;string&gt; | Comma-separated list of frame numbers of which to take screen shots  | no screenshots |
 | -sf&nbsp;&lt;string&gt;<br>&#x2011;&#x2011;ScreenshotFormat&nbsp;&lt;string&gt; | Color Space format of screenshot files. Formats are UNORM, SNORM, USCALED, SSCALED, UINT, SINT, SRGB  | Format of swapchain image |
+| -q&nbsp;&lt;bool&gt;<br>&#x2011;&#x2011;QuitOnAnyError&nbsp;&lt;bool&gt; | Quit on any error happens in replay | false |
```
Comment (davidlunarg) [inline vktrace/vktrace.md]: Better wording for the description of this option would be "Exit if an error occurs during replay".

Given the different description, maybe want to change the option to -x? But leaving it as -q would be OK.

File: vktrace/vktrace_replay/vkreplay_main.cpp [hunk 2]
```diff
@@ -111,6 +111,13 @@ vktrace_SettingInfo g_settings_info[] = {
      {&replaySettings.screenshotColorFormat},
      TRUE,
      "Color Space format of screenshot files. Formats are UNORM, SNORM, USCALED, SSCALED, UINT, SINT, SRGB"},
+    {"q",
+     "QuitOnAnyError",
+     VKTRACE_SETTING_BOOL,
+     {&replaySettings.quitOnAnyError},
+     {&replaySettings.quitOnAnyError},
+     TRUE,
+     "Quit on any error happens in replay, default is FALSE"},
```
Comment (davidlunarg) [inline vktrace/vktrace_replay/vkreplay_main.cpp]: Better wording: "Exit if an error occurs during replay, default is FALSE.

File: vktrace/vktrace_replay/vkreplay_settings.cpp [hunk 3]
```diff
@@ -86,6 +86,13 @@ vktrace_SettingInfo g_vk_settings_info[] = {
      {&s_defaultVkReplaySettings.screenshotColorFormat},
      TRUE,
      "Color Space format of screenshot files. Formats are UNORM, SNORM, USCALED, SSCALED, UINT, SINT, SRGB"},
+    {"q",
+     "quitOnAnyError",
+     VKTRACE_SETTING_BOOL,
+     {&g_vkReplaySettings.quitOnAnyError},
+     {&s_defaultVkReplaySettings.quitOnAnyError},
+     TRUE,
+     "Quit on any error happens in replay, default is FALSE"},
```
Comment (davidlunarg) [inline vktrace/vktrace_replay/vkreplay_settings.cpp]: Better wording: "Exit if an error occurs during replay, default is FALSE"

File: vktrace/vktrace.md [hunk 4]
```diff
@@ -111,6 +111,7 @@ The  `vkreplay` command-line  options are:
 | -c&nbsp;&lt;bool&gt;<br>&#x2011;&#x2011;CompatibilityMode&nbsp;&lt;bool&gt; | Enable compatibility mode - modify api calls as needed when replaying trace file created on different platform than replay platform. For example: Convert trace file memory indices to replay device memory indices. | true |
 | -s&nbsp;&lt;string&gt;<br>&#x2011;&#x2011;Screenshot&nbsp;&lt;string&gt; | Comma-separated list of frame numbers of which to take screen shots  | no screenshots |
 | -sf&nbsp;&lt;string&gt;<br>&#x2011;&#x2011;ScreenshotFormat&nbsp;&lt;string&gt; | Color Space format of screenshot files. Formats are UNORM, SNORM, USCALED, SSCALED, UINT, SINT, SRGB  | Format of swapchain image |
+| -x&nbsp;&lt;bool&gt;<br>&#x2011;&#x2011;QuitOnAnyError&nbsp;&lt;bool&gt; | Exit if an error occurs during replay | false |
```
Comment (davidlunarg) [inline vktrace/vktrace.md]: The long form of this option should be consistent with the short form of "-x".  Can you change it to "--ExitOnAnyError"? Sorry I didn't notice this when I gave you feedback on changing the short form to -x and changing the wording of the description.

This change would also affect a couple of places in the code.

Comment (justuswang) [reply vktrace/vktrace.md]: Done.

**Pull request merged**